### PR TITLE
BOAC-1468: Rewrites curated group page header in Vue

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1477

I added the [Flask-Cors](https://flask-cors.readthedocs.io/en/latest/) library to handle cross-origin requests from Vue components (localhost:8080) to APIs (localhost:5000).  GET, POST, and OPTIONS can be handled without it, but DELETE and PUT require the `Content-Type` header to be explicitly set.

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -51,6 +51,12 @@ const mutations = {
     if (indexOf > -1) {
       state.errors.splice(indexOf, 1);
     }
+  },
+  updateCuratedGroup: (state: any, data: any) => {
+    var group = state.user.myCuratedCohorts.find(
+      group => group.id === +data.id
```
Comment (johncrossman) [inline src/store.ts]: @lyttam Is that a stray '+'?

File: src/views/group/CuratedGroup.vue [hunk 2]
```diff
@@ -1,125 +1,190 @@
...
   computed: {
-    curatedGroups() {
+    pluralizeStudents() {
+      if (this.curatedGroup.students.length === 0) {
+        return '0 students';
+      } else if (this.curatedGroup.students.length === 1) {
+        return 'One student';
+      } else {
+        return this.curatedGroup.students.length + ' students';
+      }
```
Comment (johncrossman) [inline src/views/group/CuratedGroup.vue]: @lyttam Does it make sense to use the `pluralize` filter defined in `main.ts`? You might enhance it such that `length === 1` gives the nice "One" wording.

File: src/views/group/CuratedGroup.vue [hunk 3]
```diff
@@ -1,125 +1,190 @@
...
+                  class="btn-link cohort-manage-btn-link"
+                  v-on:click="enterRenameMode">
+            Rename
+          </button>
+          <span>
+            <span class="faint-text">|</span>
+            <b-btn v-b-modal="'myModal'"
+                    id="delete-cohort-button"
+                    aria-label="Delete this cohort"
+                    class="btn-link cohort-manage-btn-link">
```
Comment (johncrossman) [inline src/views/group/CuratedGroup.vue]: Minor: indentation fix needed above

**Comment by johncrossman** [comment]
Looks good, @lyttam Thank you for leading the way on `v-b-modal`.  A few minor questions posted.

File: src/views/group/CuratedGroup.vue (hunk 2 above)
Comment (lyttam) [reply src/views/group/CuratedGroup.vue]: Absolutely, thanks for pointing that out. Will amend.

**Comment by lyttam** [comment]
@johncrossman, amended with your suggestions - thanks!

**Comment by johncrossman** [comment]
Thank you @lyttam  Nicely done.

**Pull request merged**